sw/device/tests/alert_handler_lpg_clkoff_test.c

```diff
@@ -0,0 +1,743 @@
+
+    // Reset the device after last peripheral
+    // to unlock the alert handler for the next phase
+    if (test_step_cnt == (test_phase + 1) * ARRAYSIZE(kPeripherals) - 1) {
+      CHECK_DIF_OK(dif_rstmgr_software_device_reset(&rstmgr));
+    }
+  }
+
+  /* TEST PHASE #2: TEST THE FATAL ALERTS
+
+    The timeout value is set to 256
+    None of the alerts should trigger ping_timeout_alert
+    TEST:
+    enable_alerts (all_peripherals, timeout=256)
+    for_each_peripheral
+      1- wait for random time
+      2- Enable the peripheral's clock
+      3- Trigger the fatal alert of the peripheral
+      4- Disable the clock of the peripheral
+      5- Wait for 1ms
+      6- Enable the clock of the peripheral
+      7- Wait for 1ms
+      8- Confirm that fatal_alert is still there
+  */
+  if (test_step_cnt == 2 * ARRAYSIZE(kPeripherals)) {
+    alert_handler_config_all_peripherals(/* ping_timeout*/ 256);
+  }
+  while (test_step_cnt < 3 * ARRAYSIZE(kPeripherals) - 1) {
+    // Each test phase consists of ARRAYSIZE(kPeripherals) steps
+    // At every step, a specific peripheral is tested
+    test_step_cnt = flash_ctrl_testutils_counter_get(kCounterTestSteps);
+    test_phase = test_step_cnt / ARRAYSIZE(kPeripherals);
+    peri_idx = test_step_cnt - (test_phase)*ARRAYSIZE(kPeripherals);
+    // LOG_INFO("Test phase = %d, Periphereal idx = %d, test_step_cnt = %d",
+    // test_phase, peri_idx, test_step_cnt);
+
+    // Wait for a random time
+    busy_spin_micros(rnd_wait_times[peri_idx]);
+
+    // Enable the peripheral's clock
+    if (kPeripherals[peri_idx].is_hintable) {
+      CHECK_DIF_OK(dif_clkmgr_hintable_clock_set_hint(
+          &clkmgr, kPeripherals[peri_idx].clk_index, kDifToggleEnabled));
+    } else {
+      CHECK_DIF_OK(dif_clkmgr_gateable_clock_set_enabled(
+          &clkmgr, kPeripherals[peri_idx].clk_index, kDifToggleEnabled));
+    }
+
+    // Wait for 100us
+    busy_spin_micros(100 * 1);
+
+    // Trigger the fatal alert via ALERT_TEST_REG of the peripheral
+    uint32_t alert_test_reg = bitfield_bit32_write(
+        0, kPeripherals[peri_idx].fatal_alert_bit, kDifToggleEnabled);
+    mmio_region_t base_addr =
+        mmio_region_from_addr(kPeripherals[peri_idx].base);
+    mmio_region_write32(base_addr, kPeripherals[peri_idx].offset,
+                        alert_test_reg);
+
+    // Check if the alert is really triggered
+    CHECK_DIF_OK(dif_alert_handler_alert_is_cause(
+        &alert_handler, kPeripherals[peri_idx].alert_ids[0], &is_cause));
+    CHECK(is_cause, "PART #2: is_cause for alert[%d] should be 1 but we got 0",
+          kPeripherals[peri_idx].alert_ids[0]);
+
+    // Reset the alert_test register of the peripheral IP
+    alert_test_reg = bitfield_bit32_write(
+        0, kPeripherals[peri_idx].fatal_alert_bit, kDifToggleDisabled);
+    mmio_region_write32(base_addr, kPeripherals[peri_idx].offset,
+                        alert_test_reg);
```

Comment:
The alert test registers on the peripheral side should be stateless, so clearing this register is not actually needed here. One write transaction to an alert test register == one alert event transmitted to the alert handler.